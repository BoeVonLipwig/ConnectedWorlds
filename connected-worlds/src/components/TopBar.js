import React from "react";
import SelectButton from "./SelectButton";

class TopBar extends React.Component {
  clickHandler() {
    alert("Hello");
  }

  render() {
    const elems = [
      { name: "Projects", id: "showProjects" },
      { name: "Programme", id: "showSchools" },
      { name: "Collaborators", id: "showCollab" }
    ];
    const items = elems.map(elem => {
      /* items are each view toggle*/
      return <SelectButton key={elem.id} name={elem.name} id={elem.id} />;
    });
    return (
      <div id="navbar">
        <div id="controls" className="topnav">
          <a onClick={this.clickHandler}>
            <div id="view" className="">
              <h1>VIEW</h1>
              <h3 className="icon">&#9776;</h3>
            </div>
          </a>
          {items}
        </div>
        <div id="toggle">
          <div className="checkbox-round">
            <input id="showInfo" type="checkbox" defaultChecked />
            <label htmlFor="showInfo" />
          </div>
          <h2>Show Details</h2>
        </div>
      </div>
    );
  }
}

export default TopBar;
