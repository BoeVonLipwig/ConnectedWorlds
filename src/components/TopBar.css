#view h1 {
  float: left;
  font-size: 20px;
  font-weight: 500;
  padding-top: 7px;
  margin-left: 20px;
  margin-right: 0;
  margin-top: 0;
  margin-bottom: 0;
}

#view {
  padding-top: 10px;
  padding-bottom: 10px;
  height: 38px;

  color: var(--fg);
  float: left;
}

#controls > input {
  margin-left: 20px;
}

#navbar {
  border-bottom: solid 1px var(--ll);
  height: 58px;
  width: 100%;
  background-color: var(--bg);
  z-index: 2;
}

.topnav {
  position: relative;
  display: inline-block;
  background-color: var(--bg);
  color: var(--fg);
  text-transform: uppercase;
  width: 100%;
  left: 0px;
  border-bottom: solid 1px var(--ll);
}

.topnav {
  /*overflow: hidden;*/
  z-index: 6;
}

@media screen and (max-width: 700px) {
  #view:hover {
    background-color: var(--fg);
    color: var(--bg);
  }

  #view {
    width: 100%;
  }

  .topnav.responsive {
    position: relative;
    border-bottom: solid 1px var(--ll);
    z-index: 9;
  }
}

@media (max-width: 1160px) and (min-width: 0px) {
  #navbar {
    height: 115px;
    border-bottom: solid 1px var(--ll);
  }
}
