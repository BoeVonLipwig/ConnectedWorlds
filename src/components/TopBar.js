import React from "react";
import Views from "./Views";
import DetailsPane from "./DetailsPane";
import { observer } from "mobx-react";

class TopBar extends React.Component {
  clickHandler() {
    alert("Hello");
  }

  render() {
    return (
      <div id="navbar">
        <div id="controls" className="topnav">
          <a onClick={this.clickHandler}>
            <div id="view" className="">
              <h1>VIEW</h1>
              <h3 className="icon">&#9776;</h3>
            </div>
          </a>
          <Views />
        </div>
        <DetailsPane />
      </div>
    );
  }
}

export default observer(TopBar);
