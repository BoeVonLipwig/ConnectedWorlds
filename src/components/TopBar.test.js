import React from "react";
import ReactDOM from "react-dom";
import TopBar from "./TopBar";
import { shallow } from "enzyme";

it("renders topbar without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<TopBar />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders topbar correctly without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<TopBar />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});
