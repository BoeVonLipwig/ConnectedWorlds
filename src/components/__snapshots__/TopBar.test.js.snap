// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders topbar correctly without crashing 1`] = `
<div>
  <div
    id="navbar"
  >
    <div
      id="controls"
      class="topnav"
    >
      <a>
        <div
          id="view"
          class=""
        >
          <h1>
            VIEW
          </h1>
        </div>
      </a>
      <div
        class="control"
      >
        <div
          class="checkbox-round"
        >
          <input
            id="showSegment"
            type="checkbox"
            readonly=""
            checked=""
          />
          <label
            for="showSegment"
          />
        </div>
        <h2>
          Semi-Circle
        </h2>
      </div>
      <div
        class="control"
      >
        <div
          class="checkbox-round"
        >
          <input
            id="showCircles"
            type="checkbox"
            readonly=""
          />
          <label
            for="showCircles"
          />
        </div>
        <h2>
          Groups
        </h2>
      </div>
      <div
        class="control"
      >
        <div
          class="checkbox-round"
        >
          <input
            id="showCollab"
            type="checkbox"
            readonly=""
          />
          <label
            for="showCollab"
          />
        </div>
        <h2>
          Collaborators
        </h2>
      </div>
      <div
        class="control"
      >
        <div
          id="drop-down-menu"
          class=""
        >
          <div
            class="button-css"
          />
        </div>
      </div>
    </div>
    <div
      id="detailsBar"
    >
      <div
        id="toggle"
      >
        <div
          class="control"
        >
          <div
            class="checkbox-round"
          >
            <input
              id=""
              type="checkbox"
              readonly=""
              checked=""
            />
            <label
              for=""
            />
          </div>
          <h2 />
        </div>
        <h2>
          Show details
        </h2>
      </div>
      <div
        id="nodeDetails"
        class="expanded"
      >
        <div
          id="infoContainer"
          class="info"
        >
          <div
            class="container"
          >
            <em>
              Select Any Node
            </em>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
